ts/src/bingx.ts

```diff
@@ -0,0 +1,2317 @@
+            request['limit'] = limit;
+        }
+        if (market['spot']) {
+            throw new NotSupported (this.id + ' fetchOHLCV is not supported for spot markets');
+        }
+        const response = await this.swapV2PublicGetQuoteKlines (this.extend (request, params));
+        //
+        //    {
+        //        "code": 0,
+        //        "msg": "",
+        //        "data": [
+        //          {
+        //            "open": "19396.8",
+        //            "close": "19394.4",
+        //            "high": "19397.5",
+        //            "low": "19385.7",
+        //            "volume": "110.05",
+        //            "time": 1666583700000
+        //          },
+        //          ...
+        //        ]
+        //    }
+        //
+        let ohlcvs = this.safeValue (response, 'data', []);
+        if (typeof ohlcvs === 'object') {
+            ohlcvs = [ ohlcvs ];
+        }
+        return this.parseOHLCVs (ohlcvs, market, timeframe, since, limit);
+    }
+
+    parseOHLCV (ohlcv, market = undefined) {
+        //
+        //    {
+        //        "open": "19394.4",
+        //        "close": "19379.0",
+        //        "high": "19394.4",
+        //        "low": "19368.3",
+        //        "volume": "167.44",
+        //        "time": 1666584000000
+        //    }
+        //
+        return [
+            this.safeInteger (ohlcv, 'time'),
+            this.safeNumber (ohlcv, 'open'),
+            this.safeNumber (ohlcv, 'high'),
+            this.safeNumber (ohlcv, 'low'),
+            this.safeNumber (ohlcv, 'close'),
+            this.safeNumber (ohlcv, 'volume'),
+        ];
+    }
+
+    async fetchTrades (symbol: string, since: Int = undefined, limit: Int = undefined, params = {}) {
+        /**
+         * @method
+         * @name bingx#fetchTrades
+         * @description get the list of most recent trades for a particular symbol
+         * @see https://bingx-api.github.io/docs/#/spot/market-api.html#Query%20transaction%20records
+         * @see https://bingx-api.github.io/docs/#/swapV2/market-api.html#The%20latest%20Trade%20of%20a%20Trading%20Pair
+         * @param {string} symbol unified symbol of the market to fetch trades for
+         * @param {int|undefined} since timestamp in ms of the earliest trade to fetch
+         * @param {int|undefined} limit the maximum amount of trades to fetch
+         * @param {object} params extra parameters specific to the bingx api endpoint
+         * @returns {[object]} a list of [trade structures]{@link https://docs.ccxt.com/en/latest/manual.html?#public-trades}
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        let method = undefined;
+        let marketType = undefined;
+        [ marketType, params ] = this.handleMarketTypeAndParams ('fetchTrades', market, params);
+        if (marketType === 'spot') {
+            method = 'spotV1PublicGetMarketTrades';
```

Comment:
now that we have the implicit api methods we should avoid this, can we do instead something like:

```Javascript
        let response = undefined;
        if (marketType === 'spot') {
            response = await this.spotV1PublicGetMarketTrades(...);
        }
        else {
            response = await this.swapV2PublicGetQuoteTrades(...);
        }
````
⬆️ this is less error-prone